src/main/java/org/opentripplanner/api/common/RoutingResource.java

```diff
@@ -94,6 +94,10 @@
     @QueryParam("maxWalkDistance")
     protected Double maxWalkDistance;
 
+    /** The maximum distance (in meters) the user is willing to walk - specific to IBH. Defaults to unlimited. */
+    @QueryParam("maxWalkDistanceHeuristic")
+    protected Double maxWalkDistanceHeuristic;
+
```

Comment:
Great questions. We had to add a separate maximum distance specific to the heuristic, to fix a bug that was essentially an emergent property of how flag stops interacted with InterleavedBidirectionalHeuristic. I completely agree that this probably won't survive the switch to Raptor.

Here's an example of the bug. Because the walk distance is interpreted strictly, you (a) walk to a flag stop pickup, (b) immediately alight at a stop, (c) take a precomputed transfer to a different route.
![screen shot 2017-10-03 at 10 42 58 am](https://user-images.githubusercontent.com/5751667/45903547-15130400-bdb8-11e8-8e01-9cfaf3cd56fb.png)

We found that if we increased the radius of the walking search in InterleavedBidirectionalHeuristic to larger than the "maxWalkDistance" values users were selecting, it fixed the bug.

Alternatively, we could not set a maxWalkDistance and just use higher walkReluctance values, which I know has been discussed in some other contexts due to some theoretical issues with hard limiting.

Thoughts?

Reply:
Thank you for explanation, I don´t understand every detail here, especially the InterleavedBidirectionalHeuristic is a bit foggy to me, but I am happy to approve it.

Reply:
Also, in the Javadoc please expand all abbreviations like IBH.

Reply:
Thanks @abyrd. Based on this comment and the explanation at the bottom of this page, I'm going to pull out all the changes I made to InterleavedBidirectionalHeuristic. It appears the only reason I needed the extra maxWalkDistanceHeuristic value was because I removed the change setting softWalkLimiting=false -- which I shouldn't have done due to its algorithmic invalidity!

For what it's worth, as a matter of readable code, I don't like that the changes which turn off resource limiting (`softWalkLimiting=false` and `softPreTransitLimiting=false`) happen in InterleavedBidirectionalHeuristic rather than being set in RoutingRequest, where their defaults are `true`. That's pretty confusing for developers, and it means that those parameters default to `true`, invalidly, if someone uses the trivial or Euclidean heuristics. Would it cause a problem if we changed that? I can submit another PR, it's just a few lines of code.